pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4518,70 +4555,8 @@ static __always_inline u32 cgroup_skb_generic(struct __sk_buff *ctx)
 
     nethdrs hdrs = {0}, *nethdrs = &hdrs;
 
-    return CGROUP_SKB_HANDLE(family);
-}
-
-SEC("cgroup_skb/ingress")
-int cgroup_skb_ingress(struct __sk_buff *ctx)
-{
-    return cgroup_skb_generic(ctx);
-}
-
-SEC("cgroup_skb/egress")
-int cgroup_skb_egress(struct __sk_buff *ctx)
-{
-    return cgroup_skb_generic(ctx);
-}
-
-//
-// Network Protocol Events Logic
-//
-
-// when guessing by src/dst ports, declare here
-
-    #define UDP_PORT_DNS 53
-    #define TCP_PORT_DNS 53
-
-// when guessing through l7 layer, here
-
-static __always_inline int net_l7_is_http(struct __sk_buff *skb, u32 l7_off)
-{
-    #define http_min_len 7 // longest http command is "DELETE "
-
-    char http_min_str[http_min_len];
-    __builtin_memset((void *) &http_min_str, 0, sizeof(char) * http_min_len);
-
-    // load first http_min_len bytes from layer 7 in packet.
-    if (bpf_skb_load_bytes(skb, l7_off, http_min_str, http_min_len) < 0) {
-        return 0; // failed loading data into http_min_str - return.
-    }
-
-    // check if HTTP response
-    if (has_prefix("HTTP/", http_min_str, 6)) {
-        return proto_http_resp;
-    }
-
-    // clang-format off
-    // check if HTTP request
-    if (has_prefix("GET ", http_min_str, 5)    ||
-        has_prefix("POST ", http_min_str, 6)   ||
-        has_prefix("PUT ", http_min_str, 5)    ||
-        has_prefix("DELETE ", http_min_str, 8) ||
-        has_prefix("HEAD ", http_min_str, 6)) {
-        return proto_http_req;
-    }
-    // clang-format on
-
-    return 0;
-}
-
-//
-// SUPPORTED SOCKET FAMILY TYPES (inet, inet6)
-//
-
-CGROUP_SKB_HANDLE_FUNCTION(family)
```

Comment:
The family handle function was added to the initial cgroup/skb function because we needed this logic to access the bpf map and pick the event context for the packet being processed.